src/program.hpp

```diff
@@ -759,6 +765,43 @@ struct cvk_program : public _cl_program, api_object<object_magic::program> {
         return m_binary.required_work_group_size(kernel);
     }
 
+    uint32_t required_sub_group_size(std::string& kernel) const {
```

Comment:
Surely we should find a better way of doing this :). Is the information encoded in the SPIR-V module at all? If not we should teach clspv about the attribute and likely emit a **SubgroupSize** Execution Mode that we could either parse at binary loading time.

Reply:
I am not a fan of that idea. It means having to go through SPIRV-Registry SPIRV-Headers and (maybe?) SPIRV-Tools. All that to let clspv do the parsing instead of clvk? I have trouble understanding what's better that way.